src/bank.rs

```diff
@@ -277,19 +295,66 @@ impl Bank {
 
     /// Process a Transaction. This is used for unit tests and simply calls the vector Bank::process_transactions method.
     pub fn process_transaction(&self, tx: &Transaction) -> Result<()> {
-        match self.process_transactions(&[tx.clone()])[0] {
+        let rv = match self.process_transactions(&[tx.clone()])[0] {
             Err(ref e) => {
                 info!("process_transaction error: {:?}", e);
                 Err((*e).clone())
             }
             Ok(_) => Ok(()),
+        };
+        if rv.is_ok() {
+            // unlock the account if transaction is successfull
+            // bank will unlock failed transactions automatically
+            self.unlock_accounts(&[tx.clone()]);
+        }
+        rv
+    }
+    fn lock_account(
+        keys: &[Pubkey],
+        account_locks: &mut HashSet<Pubkey>,
```

Comment:
This should be the first parameter in both `lock_account` and `unlock_account`. Since you're adding all new code here, how about creating a new AccountLocks object with these as methods? No problem if the definition sits in bank.rs for now.

Reply:
close enough